tests/test_codecs/test_codecs.py

```diff
@@ -370,17 +362,66 @@ def test_invalid_metadata(store: Store) -> None:
         )
 
 
+@pytest.mark.parametrize("store", ["local", "memory"], indirect=["store"])
+def test_invalid_metadata_create_array(store: Store) -> None:
+    spath3 = StorePath(store, "invalid_order")
+    with pytest.raises(TypeError):
+        zarr.create_array(
+            spath3,
+            shape=(16, 16),
+            chunks=(16, 16),
+            dtype=np.dtype("uint8"),
+            fill_value=0,
+            filters=[
+                TransposeCodec(order="F"),  # type: ignore[arg-type]
```

Comment:
Is this testing anything more than `TransposeCodec`'s ability to correctly type-check its inputs? In which case maybe this should be covered by the codec tests?